library/game_state.py

```diff
@@ -131,15 +140,49 @@ def confirm_entry(self, term: blessed.Terminal) -> None:
 
         self.redraw_user_term(term)
 
-    def end_of_turn(self, term: blessed.Terminal) -> None:
+    def end_of_turn(self, term: blessed.Terminal, board: Board) -> None:
         """End of Turn"""
         self.current = 40
 
-        # update board
-        self.update_board = True
+        # Handle end of turn
+        working_space_location = convert_to_space_location(self.user_select_space)
+        subgrid_number = str(self.user_select_subgrid)

```

Comment:
It's probably too late for this, but can we find a single representation of this and use it everywhere? It keeps getting turned into a string and back into an int.